tools/bin/ci_credentials.sh

```diff
@@ -1,26 +1,150 @@
 #!/usr/bin/env bash
 
 . tools/lib/lib.sh
+. tools/lib/gcp-token.sh
 
 set -e
 
+COMMAND_NAME=$1
+
+# all secrets will be loaded if the second argument is not present
+CONNECTOR_FULLNAME=${2:-all}
+CONNECTOR_NAME=`echo ${CONNECTOR_FULLNAME} | rev | cut -d'/' -f1 | rev`
+
+GSM_SCOPES="https://www.googleapis.com/auth/cloud-platform"
+
+declare -A SECRET_MAP
+
+
 function write_standard_creds() {
   local connector_name=$1
   local creds=$2
   local cred_filename=${3:-config.json}
+  local source_name=${4:-github}
 
   [ -z "$connector_name" ] && error "Empty connector name"
-  [ -z "$creds" ] && error "Creds not set for $connector_name"
+  [ -z "$creds" ] && error "!!!!!Creds not set for $connector_name"
+
+  if [[ $CONNECTOR_NAME != "all" && ${connector_name} != ${CONNECTOR_NAME} ]]; then
+    return 0
+  fi
+  local key="${connector_name}#${cred_filename}"
+  [[ -z "${creds}" ]] && error "Don't find data for the connector '${key})"
+  
+  if [ -v SECRET_MAP[${key}] ]; then
+    echo "The connector '${key}' was added before"
+    return 0
+  fi
+  echo "register the secret ${key} from ${source_name}"
+  SECRET_MAP[${key}]="${creds}"
+  return 0
+}
+
+function _write_standard_creds() {
+  local connector_name=$1
+  local cred_filename=$2
+  local creds=$3
+  if jq -e . >/dev/null 2>&1 <<<${creds}; then
+      error "Failed to parse JSON for '${connector_name}' => ${cred_filename}"
+  fi
 
   if [ "$connector_name" = "base-normalization" ]; then
     local secrets_dir="airbyte-integrations/bases/${connector_name}/secrets"
   else
     local secrets_dir="airbyte-integrations/connectors/${connector_name}/secrets"
   fi
   mkdir -p "$secrets_dir"
+  echo "Saved a secret => ${secrets_dir}/${cred_filename}"
   echo "$creds" > "${secrets_dir}/${cred_filename}"
 }
 
+function save_all() {
+  for key in "${!SECRET_MAP[@]}"; do
+    local connector_name=$(echo ${key} | cut -d'#' -f1)
+    local cred_filename=$(echo ${key} | cut -d'#' -f2)
+    local creds=${SECRET_MAP[${key}]}
+    _write_standard_creds ${connector_name} ${cred_filename} "${creds}" 
+    
+  done
+  return 0
+}
+
+
+function export_github_secrets(){
```

Comment:
We should also assert this var is available
```suggestion
function export_github_secrets(){
  # We expect that all secrets injected from github are available in an env variable `SECRETS_JSON`
```